virttest/storage_migration.py

```diff
@@ -0,0 +1,74 @@
+"""
+Classes and functions to handle storage vm migration.
+
+Include features:
+1. live snapshot
+2. block mirror
+3. block stream
+4. live commit
+5. transaction
+"""
+import logging
+
+from avocado.core import exceptions
+
+
+class StorageMigrationTest(object):
+
+    """
+    KVM classes for handling common operations of storage vm migration.
+    """
+
+    def __init__(self, params, env):
+        """
+        Init the default values for live snapshot object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.params = params
+        self.env = env
+        self.vm = self.env.get_vm(self.params["main_vm"])
+
+
+class LiveSnapshotTest(StorageMigrationTest):
+
+    """
+    KVM classes for handling operations of live snapshot.
+    """
+
+    def __init__(self, params, env, device, snapshot_file, **kwargs):
+        """
+        Init the default values for live snapshot object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        :param device: the name of the device to generate the snapshot from.
+        :param snapshot_file: the target of the new image. A new file will be created.
+        :param kwargs: optional keyword arguments to pass to func
+        """
+        StorageMigrationTest.__init__(self, params, env)
+        self.device = device
+        self.snapshot_file = snapshot_file
+        self.kwargs = kwargs
+
+    def create_snapshot(self):
```

Comment:
MultihostMigration had been declared at /virttest/utils_test/qemu.py.
In contrast with libvirt/virsh, MigrationTest had been declared at ./virttest/utils_test/libvirt.py.

So, Agree with you @ waynesun09.